### PR TITLE
scripts: add upload-zynq7000 via smt2 JTAG adapter

### PR DESCRIPTION
**Pull request by maska989**
## Description
It gives access to flash zynq7000 zedboard with Jtag and Jlink

## Motivation and Context
JIRA: CI-364

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [X] Tested by hand on: armv7a9-zynq7000-zedboard.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing linter checks and tests passed.
- [X] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Comment by nalajcie** [comment]
NOTE: the `jtag` suffix should be changed to something more specific, JTAG is a protocol (also used by JLink)

File: scripts/upload-zynq7000-smt2.gdb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+#Reset USB phy
+set *((u32*)0xE000A040) = 0
+
+monitor reset
+monitor halt
+
+load
+
+mon resume
```
Comment (nalajcie) [inline scripts/upload-zynq7000-smt2.gdb:15]: maybe use full "monitor" to be consistent with the above commands

File: scripts/upload-zynq7000-smt2.gdb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+target remote localhost:3333
```
Comment (nalajcie) [inline scripts/upload-zynq7000-smt2.gdb:1]: maybe we could add single-line comment regarding the setup (eg. openocd gdb server with onboard zynq7000 Zedboard Digilent FTDI-based SMT2 JTAG)

**Review by damianloew — COMMENTED** [review]
Id' like to merge it with some additional PR in tests - addition `-jlink` suffix to script name. Then after submodules update we could run CI on PR with switch to `smt2` without any issues.

**Review by nalajcie — DISMISSED** [review]

**Review by badochov — CHANGES_REQUESTED** [review]
Wasn't able to get it to work on my setup. All config info was passed to @maska989 during a call.

**Review by nalajcie — APPROVED** [review]

File: scripts/upload-zynq7000-smt2.gdb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+set pagination on
+
+#Reset USB phy
+set *((u32*)0xE000A040) = 0
+
+monitor adapter srst pulse_width 250
+monitor adapter srst delay 250
+
+monitor reset
+monitor halt
```
Comment (badochov) [inline scripts/upload-zynq7000-smt2.gdb:15]: I'd use single command
`monitor reset halt`

**Review by badochov — COMMENTED** [review]
Newest changes work like a charm :rocket: , nitpick only

File: scripts/upload-zynq7000-smt2.gdb (hunk 3 above)
Comment (maska989) [reply scripts/upload-zynq7000-smt2.gdb:15]: I would leave that, because on error we will se exact point where monitor failed.

**Review by badochov — APPROVED** [review]

**Pull request merged**